Lib/defcon/objects/font.py

```diff
@@ -1382,11 +1408,92 @@ def _convertToFormatVersion1RoboFabData(self, libCopy):
                         finalValues.append([])
                     finalValues[-1].append(value)
                 hintData[libKey] = finalValues
-        for key, value in hintData.items():
+        for key, value in list(hintData.items()):
             if value is None:
                 del hintData[key]
         libCopy["org.robofab.postScriptHintData"] = hintData
 
+    # -----------------------------
+    # Serialization/Deserialization
+    # -----------------------------
+
+    def getDataForSerialization(self, **kwargs):
+        from functools import partial
+
+        simple_get = partial(getattr, self)
+        serialize = lambda item: item.getDataForSerialization()
+        serialized_get = lambda key: serialize(simple_get(key))
+
+        getters = (
```

Comment:
This isn't a comment on this line per se, it's more a question on the design of the getters and setters inside of the serialization methods. If I subclass an object, add an attribute tied to a user action that can be undone/redone, how would I add this to the serialization? Would I have to duplicate the code from the base class' `getDataForSerialization` and `setDataFromSerialization` and add my attribute? If so, I'd rather that there be some way to add an attribute and it's serialization helpers.